### PR TITLE
S1-ONE 30 & 34 Email Login & Logout

### PR DESCRIPTION
**Pull request by iwannabaa**
## Story / Task

see [ONE-30](https://bixlabs.atlassian.net/browse/ONE-30)

## What this PR does?

Adds functionality to login a user by email and password. Also user is able to logout

### Before 

- A user could not login to/logout from the app

### After
User is able to login and logout

- There's a signin button
- There's a login form view
- A user can use their email/password to sign in to the app
- Error messages should appear if wrong email or no password provided. Also if user does not exist.
- A user should be able to login successfully
- When logged in, user should see a logout option in user menu

### How to test / reproduce

- Click on the sign in button in the header
- Provide user email/password, the login
- Once logged in, click on user menu and click logout

### Related PRs
N/A

## Migrations

NO

Thank you.

### CODE REVIEW

File: src/app/header/header.component.ts [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
 
   ngOnInit() {
+    this.auth.isLoggedIn.subscribe((value: boolean) => this.isLoggedIn = value);
+  }
+
+  currentUser() {
+    return this.auth.getCurrentUser();
+  }
+
+  logout(): void {
```
Comment (evasquez26) [inline src/app/header/header.component.ts]: I've seen that in some cases we specify the type of data the function returns or if simply we are doing a function. I think we should be consistent. I would vote to not specify the type of the function, what do you think?

File: src/app/login/login.service.ts [hunk 2]
```diff
@@ -5,23 +5,38 @@ import { AuthRequest } from '../models/authRequest';
 import { AuthResponse } from '../models/authResponse';
 import { CONFIG } from '../core/config';
 import { Observable } from 'rxjs/Observable';
+import { AuthenticationService } from '../shared/authentication.service';
 
 @Injectable()
 export class LoginService {
 
-  constructor(private http: Http) { }
+  constructor(private http: Http, public auth: AuthenticationService) { }
```
Comment (evasquez26) [inline src/app/login/login.service.ts]: Why public?

**Comment by evasquez26** [comment]
Nevermind, just saw the other PR about the remember me functionality :sweat_smile:

File: src/app/header/header.component.ts (hunk 1 above)
Comment (iwannabaa) [inline src/app/header/header.component.ts]: Agree!

File: src/app/login/login.service.ts (hunk 2 above)
Comment (iwannabaa) [inline src/app/login/login.service.ts]: Will update :sweat_smile:

**Pull request merged**